app/bot.go

```diff
@@ -197,19 +198,29 @@ func (a *App) ConvertUserToBot(user *model.User) (*model.Bot, *model.AppError) {
 	return a.Srv.Store.Bot().Save(model.BotFromUser(user))
 }
 
-// SetBotIconImage sets LHS icon for a bot.
-func (a *App) SetBotIconImage(botUserId string, imageData *multipart.FileHeader) *model.AppError {
-	if len(*a.Config().FileSettings.DriverName) == 0 {
-		return model.NewAppError("SetBotIconImage", "api.bot.icon_image.storage.app_error", nil, "", http.StatusNotImplemented)
-	}
-
+// SetBotIconImageFromMultiPartFile sets LHS icon for a bot.
+func (a *App) SetBotIconImageFromMultiPartFile(botUserId string, imageData *multipart.FileHeader) *model.AppError {
 	file, err := imageData.Open()
 	if err != nil {
 		return model.NewAppError("SetBotIconImage", "api.bot.set_bot_icon_image.open.app_error", nil, err.Error(), http.StatusBadRequest)
 	}
 	defer file.Close()
 
-	if _, err = parseSVG(file); err != nil {
+	file.Seek(0, 0)
+	return a.SetBotIconImage(botUserId, file)
+}
+
+// SetBotIconImage sets LHS icon for a bot.
+func (a *App) SetBotIconImage(botUserId string, file io.ReadSeeker) *model.AppError {
+	if _, err := a.GetBot(botUserId, true); err != nil {
+		return err
+	}
+
+	if len(*a.Config().FileSettings.DriverName) == 0 {
```

Comment:
This check is not necessary at this level given that when you use instantiate a filestore it does check for `DriverName` and will return an appropriate error [here](https://github.com/mattermost/mattermost-server/blob/5690df9d959078bb1017dfd7e36abc4274bd890f/services/filesstore/filesstore.go#L52) if not configured.